### PR TITLE
zenith: add nvidia GPU support

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
A friend of mine pointed out zenith from nixpkgs currently does not support nvidia GPU usage stats, so i added the ability to do so.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by PixelyIon — APPROVED** [review]
I've tested this locally and it works perfectly for my GPU
image


Result of `nixpkgs-review pr 126097` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>zenith</li>
    <li>zenith-nvidia</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126097` at 58644bf6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>zenith</li>
    <li>zenith-nvidia</li>
  </ul>
</details>

**Comment by bbigras** [comment]
I got some `[ERROR zenith::graphics_nvidia] Couldn't get utilization rates: NotSupported` but it must just be because I have a very old nvidia card.

**Review by bbigras — APPROVED** [review]

File: pkgs/tools/system/zenith/default.nix [hunk 1]
```diff
@@ -11,12 +18,20 @@ rustPlatform.buildRustPackage rec {
...
 
+  cargoBuildFlags = lib.optionals nvidiaSupport [ "--features" "nvidia" ];
   cargoSha256 = "0c2mk2bcz4qjyqmf11yqhnhy4pqxr77b3c1gvr5jfmjshx4ff7v2";
 
+  nativeBuildInputs = lib.optional nvidiaSupport makeWrapper;
   buildInputs = lib.optionals stdenv.isDarwin [ IOKit ];
 
+  postInstall = lib.optionalString nvidiaSupport ''
+    wrapProgram $out/bin/zenith \
+      --suffix LD_LIBRARY_PATH : "/run/opengl-driver/lib"
```
Comment (cpcloud) [inline pkgs/tools/system/zenith/default.nix:29]: Should `/run/opengl-driver/lib` be `${addOpenGLRunpath.driverLink}/lib`?

**Comment by cpcloud** [comment]
This is awesome!

image

**Pull request merged**